src/cgutils.cpp

```diff
@@ -523,8 +523,12 @@ static Type *julia_struct_to_llvm(jl_value_t *jt, jl_unionall_t *ua, bool *isbox
                     *jl_ExecutionEngine->getDataLayout();
 #endif
                 unsigned llvm_alignment = DL.getABITypeAlignment((Type*)jst->struct_decl);
-                unsigned julia_alignment = jst->layout->alignment;
-                assert(llvm_alignment == julia_alignment);
+                unsigned julia_alignment = jl_datatype_align(jst);
+                // Check that the alignment adheres to the heap alignment.
+                // TODO: Fix alignment calculation in LLVM.
```

Comment:
Also fix the calculation on our side both in GC and in the struct declaration....